google/resource_dns_record_set.go

```diff
@@ -181,6 +173,29 @@ func resourceDnsRecordSetDelete(d *schema.ResourceData, meta interface{}) error
 
 	zone := d.Get("managed_zone").(string)
 
+	// NS records must always have a value, so we short-circuit delete
+	// this allows terraform delete to work, but may have unexpected
+	// side-effects when deleting just that record set.
+	// Unfortunately, you can set NS records on subdomains, and those
+	// CAN and MUST be deleted, so we need to retrieve the managed zone,
+	// check if what we're looking at is a subdomain, and only not delete
+	// if it's not actually a subdomain
+	var domain string
+	if d.Get("type").(string) == "NS" {
+		if domain == "" {
```

Comment:
this condition will always be true... var domain is never updated after being set.

Reply:
Domain is updated 5 lines below, on line 190.

Reply:
`domain == ""` will always be true. Nothing updates it between 183 and 185.

Reply:
Ah, yes, for some reason I was trying to reuse that request between zones.

Which makes no sense. o.o